modules/explorer/info.go

```diff
@@ -24,61 +21,130 @@ func totalCurrency(height types.BlockHeight) types.Currency {
+// block id. To find the transaction, iterate through the block.
+func (e *Explorer) Transaction(id types.TransactionID) (types.Block, types.BlockHeight, bool) {
+	height, exists := e.transactionHashes[id]
+	if !exists {
+		return types.Block{}, 0, false
+	}
+	block, exists := e.cs.BlockAtHeight(height)
+	if !exists {
+		return types.Block{}, 0, false
+	}
+	return block, height, true
+}
 
-	// No reason that consensus should broadcast a block that it
-	// doesn't have information on
-	var currentTarget types.Target
-	if e.currentBlock.ID() == e.genesisBlockID {
-		currentTarget = types.RootDepth
-	} else {
-		var exists bool
-		currentTarget, exists = e.cs.ChildTarget(e.currentBlock.ParentID)
-		if build.DEBUG {
-			if !exists {
-				panic("The state of the current block cannot be found")
-			}
-		}
+// UnlockHash returns the ids of all the transactions that contain the unlock
+// hash. An empty set indicates that the unlock hash does not appear in the
+// blockchain.
+func (e *Explorer) UnlockHash(uh types.UnlockHash) []types.TransactionID {
+	txnMap, exists := e.unlockHashes[uh]
+	if !exists || len(txnMap) == 0 {
+		return nil
+	}
+	ids := make([]types.TransactionID, 0, len(txnMap))
+	for txid := range txnMap {
+		ids = append(ids, txid)
 	}
+	return ids
+}
 
-	// Find the seen time of the block 144 ago in the list
-	matureBlockTime := e.seenTimes[(e.blockchainHeight-144)%types.BlockHeight(len(e.seenTimes))]
+// SiacoinOutputID returns all of the transactions that contain the input
+// siacoin output id. An empty set indicates that the siacoin output id does
+// not appear in the blockchain.
+func (e *Explorer) SiacoinOutputID(id types.SiacoinOutputID) []types.TransactionID {
+	txnMap, exists := e.siacoinOutputIDs[id]
+	if !exists || len(txnMap) == 0 {
+		return nil
+	}
+	ids := make([]types.TransactionID, 0, len(txnMap))
+	for txid := range txnMap {
+		ids = append(ids, txid)
+	}
+	return ids
+}
 
-	return modules.ExplorerStatus{
-		Height:              e.blockchainHeight,
-		Block:               e.currentBlock,
-		Target:              currentTarget,
-		MatureTime:          types.Timestamp(matureBlockTime.Unix()),
-		TotalCurrency:       totalCurrency(e.blockchainHeight),
-		ActiveContractCount: e.activeContractCount,
-		ActiveContractCosts: e.activeContractCost,
-		ActiveContractSize:  e.activeContractSize,
-		TotalContractCount:  e.totalContractCount,
-		TotalContractCosts:  e.totalContractCost,
-		TotalContractSize:   e.totalContractSize,
+// FileContractIDs returns all of the transactions that contain the input file
+// contract id. An empty set indicates that the file contract id does not
+// appear in the blockchain.
+func (e *Explorer) FileContractID(id types.FileContractID) []types.TransactionID {
+	txnMap, exists := e.fileContractIDs[id]
+	if !exists || len(txnMap) == 0 {
+		return nil
+	}
+	ids := make([]types.TransactionID, 0, len(txnMap))
+	for txid := range txnMap {
+		ids = append(ids, txid)
+	}
```

Comment:
I'm fairly sure you could rewrite all of these functions as:
```go
var s []sliceType
for x := range e.map[id] {
    s = append(s, x)
}
return s
```
since the callers don't distinguish between `nil` and empty, and `!exists` just means `txnMap` will be empty. But it's probably best to be explicit rather than fancy